tests/test_reply_function_sync.js

```diff
@@ -48,152 +33,199 @@ test('get with reply callback returning object', async t => {
   scope.done()
 })
 
-test('get with reply callback returning array with headers', async t => {
+test('reply with status code and function returning body as number', async t => {
   const scope = nock('http://example.test')
     .get('/')
-    .reply(() => [202, 'body', { 'x-key': 'value', 'x-key-2': 'value 2' }])
-
-  const { headers, rawHeaders } = await got('http://example.test/')
-
-  t.deepEqual(headers, {
-    'x-key': 'value',
-    'x-key-2': 'value 2',
-  })
-  t.deepEqual(rawHeaders, ['x-key', 'value', 'x-key-2', 'value 2'])
-  scope.done()
-})
-
-test('reply headers work with function', async t => {
-  const scope = nock('http://example.com')
-    .get('/')
-    .reply(200, () => 'ABC', { 'X-My-Headers': 'My custom header value' })
+    .reply(200, () => 123)
 
-  const { headers } = await got('http://example.com/')
-
-  t.equivalent(headers, { 'x-my-headers': 'My custom header value' })
+  const { body } = await got('http://example.test')
+  t.equal(body, '123')
   scope.done()
 })
 
-// Skipped because https://github.com/nock/nock/issues/1222
+// The observed behavior is that this returns a 123 status code.
+//
+// The expected behavior is that this should either throw an error or reply
+// with 200 and the JSON-stringified '[123]'.
```

Comment:
I expected the observed behavior, given the current documentation.

Is there a plan to change the expected behavior with a breaking change?

Reply:
Hmm. It seems very confusing that the 123 status code would override an also-provided 200 status code without giving an error. It doesn't seem helpful either. It seems certain that someone who wrote this did so by mistake.

Out of curiosity, is there something in the docs that makes it sound like it behaves this way? I'm game to fix it, breaking change or no. Maybe we can resolve this quickly enough to get it shipped in 11.0!

Reply:
Not from specific examples in the docs, but from an understanding that status, body, and headers can be provided two separate ways: either as separate args to `reply` or by returning an array from the callback passed to `reply`. My understanding (presumption?) was that if an array was returned from the callback, it would **override** other values. 

But if that is not explicitly the desired design; I would vote for throwing an error if both were provided because it can be confusing, especially for newcomers to the lib. I've always thought the pattern of "if its an array, then it's a special value, otherwise it's the body" was a bit too magical.

Reply:
Huh, yea, my assumption was different. It would be good to include tests of all the cases and pick the behavior that seems safest and least confusing.

Let's take a look at the behavior in the context of the changes you're proposing in #1520. We can enable these tests and make sure we've got all the bases covered.

Other than the skipped tests, do these changes look good to you?